lib/jasmine-reportportal-reporter.js

```diff
@@ -239,29 +241,39 @@ class ReportportalReporter {
     }
 
     specStarted(spec) {
-        this.startTime = this.getTime() - 1;
         const fullTestName = SpecificUtils.getFullTestName(spec);
         const promise = SpecificUtils.getCodeRef(this.currentTestFilePathIndex, fullTestName);
-
         return promise.then((codeRef) => {
-            let stepObj = this.client.startTestItem(Object.assign({
+            this.startTime = this.getTime();
+            const parent = this.getParentInfo();
+            const stepObj = this.client.startTestItem(Object.assign({
                 type: entityType.STEP,
                 description: spec.description,
+                startTime: this.startTime,
                 name: spec.description
-            }, codeRef && { codeRef }), this.tempLaunchId, this.getParentId());
+            }, codeRef && { codeRef }), this.tempLaunchId, parent && parent.tempId);
 
-            this.setParentId(stepObj.tempId);
+            this.setParentInfo({ tempId: stepObj.tempId, startTime: this.startTime });
             promiseErrorHandler(stepObj.promise);
         });
     }
 
+    getHookStartTime(hookType, parent) {
+        if (hookType === entityType.BEFORE_METHOD || hookType === entityType.BEFORE_SUITE) {
+            return Math.max(parent && parent.startTime, this.startTime - 1);
+        }
+        return this.startTime || this.getTime();
```

Comment:
I think that for `after` hooks we can return just `this.getTime()`. @KatsiarynaTatarynovich What do you think?

Reply:
I think using startTime creates a more streamlined system. If we use `this.getTime()`, the after hook may be mixed with before hook of the next test.